### PR TITLE
fix:Secret can't send to edge in SecretKeyRef #3328

### PR DESCRIPTION
**Pull request by QeelinDarly**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
This PR fixed a bug about _secret can't send to edge in SecretKeyRef #3328_. When we use secret through environment variables, the environment variables in docker cannot get the secret value.
**Which issue(s) this PR fixes**:

Fixes # Secret can't send to edge in SecretKeyRef #3328

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

```release-note

```

### CODE REVIEW

**Comment by gy95** [comment]
please sign off the commit to pass the DCO :)
ref:https://github.com/kubeedge/kubeedge/pull/3329/checks?check_run_id=4283081878

and some files need to be processed with `gofmt`

File: edge/pkg/edged/edged.go [hunk 1]
```diff
@@ -1378,14 +1378,27 @@ func (e *edged) deletePod(obj interface{}) {
...
+			if err != nil {
+				return nil, err
+			}
+			secrets = append(secrets, *secret)
 		}
-		secrets = append(secrets, *secret)
 	}
 
+
 	return secrets, nil
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go:1402]: This func is only used to get the pull secret, we can renamed it to `getPullSecretsFromMetaManager`.

File: edge/pkg/edged/edged_pods.go [hunk 2]
```diff
@@ -754,6 +755,27 @@ func (e *edged) makeEnvironmentVariables(pod *v1.Pod, container *v1.Container, p
...
+					secret, err = e.metaClient.Secrets(pod.Namespace).Get(name)
+					if err != nil {
+						if apierrors.IsNotFound(err) && optional {
+							// ignore error when marked optional
+							continue
+						}
+						return result, err
+					}
+					secrets[name] = secret
+				}
```
Comment (fisherxu) [inline edge/pkg/edged/edged_pods.go:777]: We also need to add the `optional` if here.

**Review by fisherxu — COMMENTED** [review]
Thanks for the enhancementm and would you like also to add the envfrom for secret?

File: edge/pkg/edged/edged.go (hunk 1 above)
Comment (QeelinDarly) [reply edge/pkg/edged/edged.go:1402]: Thank you for your help. I will renamed it.

**Comment by QeelinDarly** [comment]
@gy95 Thank you for your remind.

File: edge/pkg/edged/edged.go [hunk 3]
```diff
@@ -1376,16 +1376,29 @@ func (e *edged) deletePod(obj interface{}) {
...
+		for _, containers := range pod.Spec.Containers{
+			for _, env := range containers.Env{
+				secret, err := e.metaClient.Secrets(pod.Namespace).Get(env.ValueFrom.SecretKeyRef.Name)
+				if err != nil {
+					return nil, err
+				}
+				secrets = append(secrets, *secret)
+			}
+		}
+	}else{
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go:1391]: These lines are no needed.

File: edge/pkg/edged/edged_pods.go [hunk 4]
```diff
@@ -754,6 +790,30 @@ func (e *edged) makeEnvironmentVariables(pod *v1.Pod, container *v1.Container, p
...
+				runtimeVal = string(secret.Data[key])
+				if optional {
+					continue
+				}
 			}
 		}
```
Comment (fisherxu) [inline edge/pkg/edged/edged_pods.go:819]: Need to go fmted the file. :)

**Comment by QeelinDarly** [comment]
@fisherxu Can you help me to approve running workflows, please.(￣▽￣)~*

**Comment by fisherxu** [comment]
> @fisherxu Can you help me to approve running workflows, please.(￣▽￣)~*

Done.

File: edge/pkg/edged/edged_pods.go [hunk 5]
```diff
@@ -754,6 +790,30 @@ func (e *edged) makeEnvironmentVariables(pod *v1.Pod, container *v1.Container, p
...
+							continue
+						}
+						return result, err
+					}
+					secrets[name] = secret
+				}
+				runtimeVal = string(secret.Data[key])
+				if optional {
+					continue
+				}
```
Comment (fisherxu) [inline edge/pkg/edged/edged_pods.go:816]:
```suggestion
				runtimeValBytes, ok := secret.Data[key]
				if !ok {
					if optional {
						continue
					}
					return result, fmt.Errorf("couldn't find key %v in Secret %v/%v", key, pod.Namespace, name)
				}
				runtimeVal = string(runtimeValBytes)
```
Let's keep same with K8s.

**Review by fisherxu — COMMENTED** [review]
/approve
/lgtm
Thanks for the fixing!

**Pull request merged**